### PR TITLE
Softmax: add weighted-sum normalization

### PR DESCRIPTION
**Pull request by madamczykhabana**
Supporting PR for https://github.com/HabanaAI/vllm-hpu-extension/pull/10

### CODE REVIEW

File: vllm/worker/hpu_model_runner.py [hunk 1]
```diff
@@ -1226,7 +1241,7 @@ def trim_attn_metadata(self, metadata: AttentionMetadata) -> object:
         attention_metadata = subtuple(metadata, 'TrimmedAttentionMetadata', [
             'attn_bias', 'seq_lens_tensor', 'block_list', 'block_mapping',
             'block_usage', 'slot_mapping', 'is_prompt', 'block_indices',
-            'block_offsets'
+            'block_offsets', 'block_scales'
         ])
         return attention_metadata
```
Comment (michalkuligowski) [inline vllm/worker/hpu_model_runner.py:1246]: cpu test returned issue in line 301 of this file:

RuntimeError: Class values must be non-negative.

Comment (madamczykhabana) [reply vllm/worker/hpu_model_runner.py:1258]: Yeah, it turns out cpu doesn't allow out-of-bounds class values. I've added a fix for it in: https://github.com/HabanaAI/vllm-fork/pull/378/commits/a05fffca2eb53e8c6d2fe1deed45d73160195771

File: vllm/worker/hpu_model_runner.py [hunk 2]
```diff
@@ -298,9 +298,19 @@ def _set_block_mapping(self, metadata, batch_size, device, dtype):
         mask = mask >= metadata.block_usage.unsqueeze(-1)
         attn_bias = (torch.zeros_like(mask, dtype=dtype).masked_fill_(
             mask, -math.inf))
-        block_mapping = torch.nn.functional.one_hot(
-            metadata.block_mapping.to(torch.long),
-            num_classes=batch_size).to(dtype)
+        if 'cpu' in str(device).lower():
```
Comment (michalkuligowski) [inline vllm/worker/hpu_model_runner.py:301]: maybe you can use 'not is_hpu()' from vllm.utils?

**Review by michalkuligowski — APPROVED** [review]

File: vllm/worker/hpu_model_runner.py (hunk 2 above)
Comment (madamczykhabana) [reply vllm/worker/hpu_model_runner.py]: Hmm, I guess is_fake_hpu would be better here as we want this code to run with fake_hpu_mode and afaik is_hpu returns True when running in fake mode.

**Review by michalkuligowski — APPROVED** [review]

**Pull request merged**